packages/sol-compiler/src/utils/compiler.ts

```diff
@@ -117,19 +119,13 @@ export function parseDependencies(contractSource: ContractSource): string[] {
 
 /**
  * Compiles the contracts and prints errors/warnings
- * @param resolver Resolver
  * @param solcInstance Instance of a solc compiler
  * @param standardInput Solidity standard JSON input
  */
-export function compile(
-    resolver: Resolver,
-    solcInstance: solc.SolcInstance,
-    standardInput: solc.StandardInput,
-): solc.StandardOutput {
+export function compile(solcInstance: solc.SolcInstance, standardInput: solc.StandardInput): solc.StandardOutput {
     const standardInputStr = JSON.stringify(standardInput);
     const standardOutputStr = solcInstance.compileStandardWrapper(standardInputStr, importPath => {
-        const sourceCodeIfExists = resolver.resolve(importPath);
-        return { contents: sourceCodeIfExists.source };
+        throw Error('Used callback. All sources should be resolved beforehand.');
```

Comment:
`throw new Error()`
AFAIK, there is some subtle difference and you don't see that form in the wild very often.